src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,44 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name) {
+    this.health = 100;
+    this.name = name;
+  }
+
+  applyDamage(healthPoints) {
+    this.health -= healthPoints;
+
+    if (this.health < 0) {
+      Animal.removeDead();
+    }
+  }
 }
 
+Animal.alive = [];
+Animal.removeDead = () => {
```

Comment:
[FUNCTION NAMING] - The name 'removeDead' implies that the function will remove dead animals, but it's not immediately clear that it's linked to the Animal.alive array. Consider renaming the function to more accurately reflect its purpose, such as 'removeDeadFromAlive', to improve code readability and maintainability.